xikolo-common/Data/Helper/CourseHelper.swift

```diff
@@ -13,23 +13,34 @@ class CourseHelper {
 
     static fileprivate let entity = NSEntityDescription.entity(forEntityName: "Course", in: CoreDataHelper.managedContext)!
 
-    static func getAllCoursesRequest() -> NSFetchRequest<NSFetchRequestResult> {
+    static func getGenericCoursesRequest() -> NSFetchRequest<NSFetchRequestResult> {
         let request = NSFetchRequest<NSFetchRequestResult>(entityName: "Course")
-        request.predicate = NSPredicate(format: "external_int != true")
         let startDateSort = NSSortDescriptor(key: "start_at", ascending: false)
         request.sortDescriptors = [startDateSort]
         return request
     }
 
+    static func getAllCoursesRequest() -> NSFetchRequest<NSFetchRequestResult> {
+        let request = getGenericCoursesRequest()
+        request.predicate = NSPredicate(format: "external_int != true")
+        return request
+    }
+
+    static func getFindCoursesRequest() -> NSFetchRequest<NSFetchRequestResult> {
```

Comment:
i would like a naming that is related to what is does internally, not to the user facing function